app/src/main/java/org/simple/clinic/search/PatientSearchScreenController.kt

```diff
@@ -76,21 +90,117 @@ class PatientSearchScreenController @Inject constructor(
         }
   }
 
+  private fun validateQuery(): ObservableTransformer<UiEvent, UiEvent> {
+    return ObservableTransformer { events ->
+      val nameChanges = events
+          .ofType<SearchQueryNameChanged>()
+          .map { it.name.trim() }
+
+      val ageChanges = events
+          .ofType<SearchQueryAgeChanged>()
+          .map { it.ageString.trim() }
+
+      val dateOfBirthChanges = events
+          .ofType<SearchQueryDateOfBirthChanged>()
+          .map { it.dateOfBirth.trim() }
+
+      val validationErrors = events.ofType<SearchClicked>()
+          .withLatestFrom(nameChanges, ageChanges, dateOfBirthChanges) { _, name, age, dob -> Triple(name, age, dob) }
+          .map { (name, age, dateOfBirth) ->
+            val errors = mutableListOf<PatientSearchValidationError>()
+
+            if (dateOfBirth.isNullOrBlank() && age.isNullOrBlank()) {
```

Comment:
I feel a `when` clause here might be better than a if-else-if clause. What do you think?